src/settings.js

```diff
@@ -166,31 +166,56 @@ const settings = {
         await match.accessible.ButtonBarButton('Done').tap();
     },
     /**
-     * @name settings.CorrectionRange
-     * @param {corrections} corrections e.g. { range: { min: '80', max: '150' } });
-     * corrections:{
-     *    range:{ min: string, max: string,},
-     *     preMeal:{min: string, max: string},
-     *     workout:{min: string, max: string}
-     * },
+     * @name settings.CorrectionRanges
+     * @param {object} ranges e.g. [{ time: '12:00 AM', min: '80', max: '150' }];
      */
-    async CorrectionRange(corrections) {
-        console.log('corrections: ', corrections);
+    async CorrectionRanges(ranges) {
+        const correctionRangePickerColumns = {
+            Units: 5,
+            MaximumValue: 4,
+            Separator: 3,
+            MinimumValue: 2,
+            Time: 1,
+        };
+
         await match.accessible.ButtonBarButton('Settings').tap();
         await match.accessible.Text('Correction Range').tap();
         await match.accessible.ButtonBarButton('Add').tap();
 
-        if (corrections.range) {
-            console.log('corrections.range: ', corrections.range);
-            //TODO: atIndex!! need a better way
-            // await match.accessible.Label('12:00 AM').atIndex(0).tap();
-            // await match.accessible.Label(`${corrections.range.max}`).atIndex(1).tap();
-            // await match.accessible.Label(`${corrections.range.min}`).atIndex(0).tap();
-            // await match.accessible.Label('Save').tap();
+        if (ranges) {
+            let pickerItemIndex = 1;
+            for (let index = 0; index < ranges.length; index++) {
+                const range = ranges[index];
+                await match.accessible.Label(`${range.time}`).atIndex(0).tap();
+                await match.accessible.PickerItem(pickerItemIndex, `${range.max}`).tap();
+                await match.accessible.PickerItem(pickerItemIndex, `${range.min}`).atIndex(correctionRangePickerColumns.MinimumValue).tap();
+                pickerItemIndex++;
+            }
         }
-        if (corrections.preMeal) {
-            console.log('corrections.preMeal: ', corrections.preMeal);
+        await match.accessible.Label('Save').tap();
+        await match.accessible.BackButton('Settings').tap();
+        await match.accessible.ButtonBarButton('Done').tap();
+    },
+    /**
+     * @name settings.CorrectionRangeOverride
+     * @param {object} override e.g. { min: '80', max: '150' };
+     */
+    async PreMealCorrectionRange(override) {
+        const glucoseRangeOverridePickerColumns = {
+            Label: 1,
+            MinimumValue: 2,
+            Separator: 3,
+            MaximumValue: 4,
+            Units: 5,
+        };
+        await match.accessible.ButtonBarButton('Settings').tap();
```

Comment:
This assumes that you are at the main page and are navigating into the `Settings` page. Might be quicker to assume you are at the `Settings` page and navigate from there. That way, you can change a whole bunch of settings at once. Something like, go to settings, change setting 1, change setting 2, change setting 3, save.

Reply:
will address next sprint in apply of default settings